OpenRA.Mods.Common/Traits/Sound/AnnounceOnSeen.cs

```diff
@@ -18,18 +20,35 @@ public class AnnounceOnSeenInfo : ITraitInfo
 		[Desc("Should there be a radar ping on enemies' radar at the actor's location when they see him")]
 		public readonly bool PingRadar = false;
 
-		public readonly string Notification = "EnemyUnitSighted";
+		public readonly string Notification = null;
 
-		public object Create(ActorInitializer init) { return new AnnounceOnSeen(this); }
+		public object Create(ActorInitializer init) { return new AnnounceOnSeen(init.Self, this); }
 	}
 
-	public class AnnounceOnSeen
+	public class AnnounceOnSeen : INotifyDiscovered
 	{
 		public readonly AnnounceOnSeenInfo Info;
 
-		public AnnounceOnSeen(AnnounceOnSeenInfo info)
+		readonly Lazy<RadarPings> radarPings;
+
+		public AnnounceOnSeen(Actor self, AnnounceOnSeenInfo info)
 		{
 			Info = info;
+			radarPings = Exts.Lazy(() => self.World.WorldActor.Trait<RadarPings>());
+		}
+
+		public void OnDiscovered(Actor self, Player discoverer, bool playNotification)
+		{
+			if (!playNotification || discoverer != self.World.RenderPlayer)
+				return;
+
+			// Audio notification
+			if (discoverer != null && !string.IsNullOrEmpty(Info.Notification))
```

Comment:
I think you don't need the `discoverere != null` check here as you check for `discoverer != self.World.RenderPlayer` above.